blackbox.cpp

```diff
@@ -113,6 +123,32 @@ class InitCleaner {
   bool disarm_ = false;
 };
 
+// Utility class for signal-safe RAII locking
+class SignalSafeLock {
+ public:
+  static std::optional<SignalSafeLock> grabLock(std::mutex &mtx) {
```

Comment:
Given this class already reads from global `lock_taken`, maybe it could also read from global `lock`?